server/src/main/java/io/druid/curator/CuratorModule.java

```diff
@@ -91,6 +101,54 @@ public void stop()
     return framework;
   }
 
+  private CuratorFrameworkFactory.Builder newCuratorBuilder(CuratorConfig config, ExhibitorConfig exConfig)
+  {
+    CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder();
+    Optional<EnsembleProvider> provider = newEnsembleProvider(exConfig, config.getZkHosts());
+    if (provider.isPresent()) {
+      return builder.ensembleProvider(provider.get());
+    }
+    return builder.connectString(config.getZkHosts());
```

Comment:
under the hood this just does a ` new FixedEnsembleProvider(connectString)`, can `newEnsembleProvider` be modified to use `FixedEnsembleProvider` if no appropriate `ExhibitorConfig ` stuff is found